gradle-plugin/configure/build.gradle.kts

```diff
@@ -1,28 +1,28 @@
-import com.novoda.gradle.release.PublishExtension
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 /// The plugin version number – change this to match whatever your tag will be
-version = "0.5.0"
 group = "com.automattic.android"
+version = "0.6.0"
 
 buildscript {
     repositories {
-        jcenter()
-        maven {
-            setUrl("https://plugins.gradle.org/m2/")
-        }
+        maven { url = uri("https://a8c-libs.s3.amazonaws.com/android") }
```

Comment:
For my own information: what's the difference between `setUrl(…)` and `url = uri(…)`?

Reply:
_Disclaimer: I've started learning about Kotlin DSL only about 2 months ago, and it's been a painful process so far. I am more used to Groovy syntax, but even then I am not comfortable with a lot of stuff. @AliSoftware This goes for a lot of other stuff (I'll stop saying it from here after), so take everything I say with a bit of salt :)_

I don't think there is any difference, because essentially both of them are just setting [this property](https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.repositories.MavenArtifactRepository.html#org.gradle.api.artifacts.repositories.MavenArtifactRepository:url). [`setUrl` seems to be just a shorthand that accepts a `uri`](https://github.com/gradle/gradle/blob/master/subprojects/core-api/src/main/java/org/gradle/api/artifacts/repositories/MavenArtifactRepository.java#L39-L47).

I typically try to follow the syntax in Gradle documentation, and I've used this syntax in [publish-to-s3-gradle-plugin project](https://github.com/Automattic/publish-to-s3-gradle-plugin), so I just copied it over. [Here is an example of this syntax usage in the docs](https://docs.gradle.org/current/userguide/declaring_repositories.html#sec:declaring_multiple_repositories).

Hope that helps!

Reply:
It does!

And that also answers my unasked question about "what is this `.kts` extension; I understand now that you can write Gradle files either using Groovy (`xxx.gradle`) – which is what I've mostly been seeing in the past – but can now also choose to write them in Kotlin (`xxx.gradle.kts`), which is what the new trend seems to be going towards I guess. Makes more sense as to why I have been seeing 2 syntaxes now! 👍

Reply:
You may also find this [Gradle Kotlin DSL Primer documentation](https://docs.gradle.org/current/userguide/kotlin_dsl.html) helpful in the future.

I don't know if one is better than the other, not yet anyway, but we are giving the Kotlin DSL a try in these smaller projects to understand its cost/benefit comparison. It also forces me to learn about what's going on under the hood instead of relying on a bunch of shorthands.